pkg/lint/lint.go

```diff
@@ -208,15 +208,42 @@ func (l *Linter) LintAsset(ctx context.Context, rootPath string, pipelineDefinit
 		rules = slices.Concat([]Rule{}, rules, assetRules)
 	}
 
-	// now the actual validation starts
-	for _, rule := range rules {
-		issues, err := rule.ValidateAsset(ctx, assetPipeline, asset)
-		if err != nil {
-			return nil, err
+	// Check if asset has URI dependencies to determine if cross-pipeline rules should run
+	hasURIDeps := false
+	for _, upstream := range asset.Upstreams {
+		if upstream.Type == "uri" {
+			hasURIDeps = true
+			break
 		}
+	}
 
-		if len(issues) > 0 {
-			pipelineResult.Issues[rule] = issues
+	// now the actual validation starts
+	for _, rule := range rules {
+		levels := rule.GetApplicableLevels()
+		if slices.Contains(levels, LevelCrossPipeline) {
+			// Cross-pipeline rules are only included when asset has URI dependencies
+			if !hasURIDeps {
+				continue
+			}
+			issues, err := rule.ValidateCrossPipeline(ctx, pipelines)
+			if err != nil {
+				return nil, err
+			}
+			// Only include issues for this asset
+			for _, issue := range issues {
+				if issue.Task == asset {
+					pipelineResult.Issues[rule] = append(pipelineResult.Issues[rule], issue)
+				}
+			}
+		} else if slices.Contains(levels, LevelAsset) {
```

Comment:
you may want to remove the `else` here.

If a rule declares both cross pipeline and asset levels, only cross pipeline validation will be run against it.